### ISSUE TITLE
Invalid LS response

### ISSUE THREAD

**Issue by akurtakov** [mads-hartmann/bash-language-server#101]
Relevant Request and replies are:
```console
LSP4E to org.eclipse.shelled.ls:{"jsonrpc":"2.0","id":"4","method":"textDocument/documentSymbol","params":{"textDocument":{"uri":"file:///home/akurtakov/work/runtime-EclipseApplication/test/test.sh"}}}
org.eclipse.shelled.ls to LSP4E:{"jsonrpc":"2.0","id":"4","result":[{"name":"add_a_user","kind":12,"location":{"uri":"file:///home/akurtakov/work/runtime-EclipseApplication/test/test.sh","range":{"start":{"line":3,"character":0},"end":{"line":14,"character":1}}}},{"name":"USER","location":{"uri":"file:///home/akurtakov/work/runtime-EclipseApplication/test/test.sh","range":{"start":{"line":5,"character":2},"end":{"line":5,"character":9}}},"containerName":"add_a_user"},{"name":"PASSWORD","location":{"uri":"file:///home/akurtakov/work/runtime-EclipseApplication/test/test.sh","range":{"start":{"line":6,"character":2},"end":{"line":6,"character":13}}},"containerName":"add_a_user"},{"name":"COMMENTS","location":{"uri":"file:///home/akurtakov/work/runtime-EclipseApplication/test/test.sh","range":{"start":{"line":9,"character":2},"end":{"line":9,"character":13}}},"containerName":"add_a_user"}]}
```
the file for which it happens is:
```sh
#!/bin/sh
# A simple script with a function...

add_a_user()
{
  USER=$1
  PASSWORD=$2
  shift; shift;
  # Having shifted twice, the rest is now comments ...
  COMMENTS=$@
  echo "Adding user $USER ..."
  echo useradd -c "$COMMENTS" $USER
  echo passwd $USER $PASSWORD
  echo "Added user $USER ($COMMENTS) with pass $PASSWORD"
}

###
# Main body of script starts here
###
echo "Start of script..."
add_a_user bob letmein Bob Holness the presenter
add_a_user fred badpassword Fred Durst the singer
add_a_user bilko worsepassword Sgt. Bilko the role model
echo "End of script..."

```

The issue is that for USER, PASSWORD and COMMENTS variables "kind" attribute is missing but it's mandatory according to spec.

**Comment by skovhus** [comment]
Thanks for reporting this. This is actually an issue with the underlying grammar. 

Can you create an issue here: https://github.com/tree-sitter/tree-sitter-bash ?

Thanks.

**Comment by akurtakov** [comment]
I'm not sure how exaclty tree-sitter-bash is used for this so can you please either:
* File the issue yourself so I don't end up creating some mess :)
* Point me to bash-language-server place using it with some guidance so I can come up with proper explanation (maybe even small repro) for tree-sitter-bash

**Comment by maxbrunsfeld** [comment]
> This is actually an issue with the underlying grammar.

👋 I just saw this issue referencing https://github.com/tree-sitter/tree-sitter-bash/issues/42. This code seems like it parsers totally fine.

<details>
<summary>Output of `tree-sitter parse the-file.sh`</summary>

```cli
(program [0, 0] - [25, 3]
  (command [0, 0] - [25, 3]
    (command_name [0, 0] - [25, 3]
      (command_substitution [0, 0] - [25, 3]
        (command [0, 1] - [25, 2]
          (command_name [0, 1] - [25, 2]
            (command_substitution [0, 1] - [25, 2]
              (command [0, 2] - [25, 1]
                (command_name [0, 2] - [25, 1]
                  (command_substitution [0, 2] - [25, 1]
                    (command [0, 3] - [0, 8]
                      (command_name [0, 3] - [0, 8]
                        (word [0, 3] - [0, 8])))
                    (comment [1, 0] - [1, 9])
                    (comment [2, 0] - [2, 36])
                    (function_definition [4, 0] - [15, 1]
                      (word [4, 0] - [4, 10])
                      (compound_statement [5, 0] - [15, 1]
                        (variable_assignment [6, 2] - [6, 9]
                          (variable_name [6, 2] - [6, 6])
                          (simple_expansion [6, 7] - [6, 9]
                            (variable_name [6, 8] - [6, 9])))
                        (variable_assignment [7, 2] - [7, 13]
                          (variable_name [7, 2] - [7, 10])
                          (simple_expansion [7, 11] - [7, 13]
                            (variable_name [7, 12] - [7, 13])))
                        (command [8, 2] - [8, 7]
                          (command_name [8, 2] - [8, 7]
                            (word [8, 2] - [8, 7])))
                        (command [8, 9] - [8, 14]
                          (command_name [8, 9] - [8, 14]
                            (word [8, 9] - [8, 14])))
                        (comment [9, 2] - [9, 54])
                        (variable_assignment [10, 2] - [10, 13]
                          (variable_name [10, 2] - [10, 10])
                          (simple_expansion [10, 11] - [10, 13]
                            (special_variable_name [10, 12] - [10, 13])))
                        (command [11, 2] - [11, 30]
                          (command_name [11, 2] - [11, 6]
                            (word [11, 2] - [11, 6]))
                          (string [11, 7] - [11, 30]
                            (simple_expansion [11, 20] - [11, 25]
                              (variable_name [11, 21] - [11, 25]))))
                        (command [12, 2] - [12, 35]
                          (command_name [12, 2] - [12, 6]
                            (word [12, 2] - [12, 6]))
                          (word [12, 7] - [12, 14])
                          (word [12, 15] - [12, 17])
                          (string [12, 18] - [12, 29]
                            (simple_expansion [12, 19] - [12, 28]
                              (variable_name [12, 20] - [12, 28])))
                          (simple_expansion [12, 30] - [12, 35]
                            (variable_name [12, 31] - [12, 35])))
                        (command [13, 2] - [13, 29]
                          (command_name [13, 2] - [13, 6]
                            (word [13, 2] - [13, 6]))
                          (word [13, 7] - [13, 13])
                          (simple_expansion [13, 14] - [13, 19]
                            (variable_name [13, 15] - [13, 19]))
                          (simple_expansion [13, 20] - [13, 29]
                            (variable_name [13, 21] - [13, 29])))
                        (command [14, 2] - [14, 57]
                          (command_name [14, 2] - [14, 6]
                            (word [14, 2] - [14, 6]))
                          (string [14, 7] - [14, 57]
                            (simple_expansion [14, 19] - [14, 24]
                              (variable_name [14, 20] - [14, 24]))
                            (simple_expansion [14, 26] - [14, 35]
                              (variable_name [14, 27] - [14, 35]))
                            (simple_expansion [14, 47] - [14, 56]
                              (variable_name [14, 48] - [14, 56]))))))
                    (comment [17, 0] - [17, 3])
                    (comment [18, 0] - [18, 33])
                    (comment [19, 0] - [19, 3])
                    (command [20, 0] - [20, 25]
                      (command_name [20, 0] - [20, 4]
                        (word [20, 0] - [20, 4]))
                      (string [20, 5] - [20, 25]))
                    (command [21, 0] - [21, 48]
                      (command_name [21, 0] - [21, 10]
                        (word [21, 0] - [21, 10]))
                      (word [21, 11] - [21, 14])
                      (word [21, 15] - [21, 22])
                      (word [21, 23] - [21, 26])
                      (word [21, 27] - [21, 34])
                      (word [21, 35] - [21, 38])
                      (word [21, 39] - [21, 48]))
                    (command [22, 0] - [22, 49]
                      (command_name [22, 0] - [22, 10]
                        (word [22, 0] - [22, 10]))
                      (word [22, 11] - [22, 15])
                      (word [22, 16] - [22, 27])
                      (word [22, 28] - [22, 32])
                      (word [22, 33] - [22, 38])
                      (word [22, 39] - [22, 42])
                      (word [22, 43] - [22, 49]))
                    (command [23, 0] - [23, 56]
                      (command_name [23, 0] - [23, 10]
                        (word [23, 0] - [23, 10]))
                      (word [23, 11] - [23, 16])
                      (word [23, 17] - [23, 30])
                      (word [23, 31] - [23, 35])
                      (word [23, 36] - [23, 41])
                      (word [23, 42] - [23, 45])
                      (word [23, 46] - [23, 50])
                      (word [23, 51] - [23, 56]))
                    (command [24, 0] - [24, 23]
                      (command_name [24, 0] - [24, 4]
                        (word [24, 0] - [24, 4]))
                      (string [24, 5] - [24, 23]))))))))))))

```
</details>

---

@skovhus Could you clarify what problem there is with the grammar? In the issue description, @akurtakov is saying that `bash-language-server` is sending an improperly-structured JSON response to the client. This seems like it has nothing to do with the grammar.

In general, I'm 100% on board with you folks asking users to open issues on `tree-sitter-bash` when they see parsing problems: it's very helpful for me to get this feedback. But in this case, asking @akurtakov to open a `tree-sitter-bash` issue doesn't make sense to me. Before you ask users to do this, can you run `tree-sitter parse` on the file, just to quickly assess if there's something wrong with the parser's output?

**Comment by skovhus** [comment]
@maxbrunsfeld thanks. I’m glad you find us redirect relevant parser issue valuable, but I see that I might have been too eager to redirect the issue here. ;)

I’ll make sure we run the parser before creating issues. 

And thanks for all your hard work!

**Comment by rcjsuen** [comment]
Thanks for the sample Bash file, @akurtakov. I've opened #110 which should hopefully fix this particular problem.

**Comment by skovhus** [comment]
Will keep this open until we release the fix. Thanks @rcjsuen

**Comment by skovhus** [comment]
Thank you for fixing this. This was released as version 1.5.6.

Upgrade by running:
`npm install -g bash-language-server`
